Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Io_image_plugin.cpp

```diff
@@ -816,17 +825,94 @@ Io_image_plugin::load(QFileInfo fileinfo) {
+                             raw_dialog.spacing_x->value(),
+                             raw_dialog.spacing_y->value(),
+                             raw_dialog.spacing_z->value(),
+                             raw_dialog.offset->value(),
+                             raw_dialog.image_word_size(),
+                             raw_dialog.image_word_kind(),
+                             raw_dialog.image_sign())
+             ){
+            switch(raw_dialog.image_word_kind())
+            {
+            case WK_FLOAT:
+              is_gray = true;
+              if(raw_dialog.image_word_size() == 8)
+              {
+                float *f_data = (float*)ImageIO_alloc(image->xdim()*image->ydim()*image->zdim()*sizeof(float));
+                double* d_data = (double*)(image->data());
+                //convert image from double to float
+                for(std::size_t x = 0; x<image->xdim(); ++x)
+                  for(std::size_t y = 0; y<image->ydim(); ++y)
+                    for(std::size_t z = 0; z<image->zdim(); ++z)
+                    {
+                      std::size_t i =(z * image->ydim() + y) * image->xdim() + x;
+                      f_data[i] =(float)d_data[i];
+                    }
+                ImageIO_free(d_data);
+                image->image()->data = (void*)f_data;
+                image->image()->wdim = 4;
+              }
+              break;
+            case WK_FIXED:
+            {
+              float *f_data = (float*)ImageIO_alloc(image->xdim()*image->ydim()*image->zdim()*sizeof(float));
+              switch(raw_dialog.image_word_size())
+              {
+              case 2:
+              {
+                is_gray = true;
+                //convert image from short to char
+                short* s_data = (short*)(image->data());
+                for(std::size_t x = 0; x<image->xdim(); ++x)
+                  for(std::size_t y = 0; y<image->ydim(); ++y)
+                    for(std::size_t z = 0; z<image->zdim(); ++z)
+                    {
+                      std::size_t i =(z * image->ydim() + y) * image->xdim() + x;
+                      f_data[i] =(char)s_data[i];
+                    }
+                ImageIO_free(s_data);
+                image->image()->data = (void*)f_data;
+                image->image()->wdim = 4;
+                image->image()->wordKind = WK_FLOAT;
+                break;
+              }
+              case 4:
+              {
+                is_gray = true;
+                //convert image from int to char
+                int* i_data = (int*)(image->data());
+                for(std::size_t x = 0; x<image->xdim(); ++x)
+                  for(std::size_t y = 0; y<image->ydim(); ++y)
+                    for(std::size_t z = 0; z<image->zdim(); ++z)
+                    {
+                      std::size_t i =(z * image->ydim() + y) * image->xdim() + x;
+                      f_data[i] =(int)i_data[i];
+                    }
+                ImageIO_free(i_data);
```

Comment:
Is not there a way to factorize all those similar loops in a member function templated by the input type and the output type?